python-spec/src/somacore/query/query.py

```diff
@@ -434,6 +501,81 @@ def _axisp_inner(
         joinids = getattr(self._joinids, axis.value)
         return axisp[layer].read((joinids, joinids))
 
+    def _axism_inner(
+        self,
+        axis: "_Axis",
+        layer: str,
+    ) -> data.SparseRead:
+        key = axis.value + "m"
+
+        if key not in self._ms:
+            raise ValueError(f"Measurement does not contain {key} data")
+
+        axism = self._ms.obsm if axis is _Axis.OBS else self._ms.varm
+        if not (layer and layer in axism):
+            raise ValueError(f"Must specify '{key}' layer")
+
```

Comment:
_axisp_inner contains an additional check for the type of the object (SparseNDArray). is there a reason to not do a check here, given that the subsequent code requires that it be a SparseNDArray?

Reply:
Good call out. I originally removed it because I was planning to support `DenseNDArray`s as well, but we ended up not doing it. Adding it back.